### PR TITLE
feat: new sharebuttons

### LINKED ISSUE #98: LinkedIn share does not work
### Current Behavior

Screenshot_20240302-081715~2

### Reproduction / Steps To Reproduce

1. Go to https://stadtgeschichtebasel.ch/blog/buchvernissage-band-1-4-der-stadt-geschichte-basel
2. Click on LinkedIn share Button
3. See error

### Anything else?

- https://github.com/rodneylab/sveltekit-share-buttons
- https://rodneylab.com/svelte-share-buttons/

### PR DESCRIPTION
**Pull request by maehr**
# Pull request

## Proposed changes

- Fix #98 
- Replace dependency
- Better styling

## Types of changes

- [ ] New feature (non-breaking change which adds functionality).
- [x] Enhancement (non-breaking change which enhances functionality)
- [ ] Bug Fix (non-breaking change which fixes an issue).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).

## Checklist

- [x] I have read the **[README](./README.md)** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Summary by CodeRabbit

- **New Features**
	- Introduced a unified `ShareButtons` component for easy content sharing across email, Facebook, Twitter, WhatsApp, and LinkedIn. Utilizes the Web Share API for enhanced sharing capabilities on supported browsers.
- **Refactor**
	- Updated the blog post layout for improved content presentation.
	- Simplified social media sharing by replacing specific share button components with a generic `ShareButtons` component across all blog posts.

### CODE REVIEW

**Review by koilebeit — COMMENTED** [review]
- On Firefox I only see one button where you can't read the font either:

image

- On Brave there are all buttons, but on hover the buttons almost disappear

**Comment by maehr** [comment]
> * On Firefox I only see one button where you can't read the font either:
> 
> image
> 
> * On Brave there are all buttons, but on hover the buttons almost disappear

I did not check for Firefox, on Chrome I did experience the same issue with the buttons. I am not yet sure what causes it...

**Comment by mtwente** [comment]
I changed the button styles so they do not disappear (fill up with white) on hover. I can see the buttons in Chrome and Firefox just fine.

Additionally, I made them smaller and changed the colour of the logos to secondary (yellow) on hover.

smaller buttons with hover-colouring

**Comment by maehr** [comment]
> I changed the button styles so they do not disappear (fill up with white) on hover. I can see the buttons in Chrome and Firefox just fine.
> 
> Additionally, I made them smaller and changed the colour of the logos to secondary (yellow) on hover.
> 
> smaller buttons with hover-colouring

I cannot see your changes. (You need to use this https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request)
I pushed some of my own.

File: src/lib/components/ShareButtons.svelte [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+					><!--!Font Awesome Free 6.5.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free Copyright 2024 Fonticons, Inc.--><title
+						>Facebook</title
+					><path
+						d="M80 299.3V512H196V299.3h86.5l18-97.8H196V166.9c0-51.7 20.3-71.5 72.7-71.5c16.3 0 29.4 .4 37 1.2V7.9C291.4 4 256.4 0 236.2 0C129.3 0 80 50.5 80 159.4v42.1H14v97.8H80z"
+					/></svg
+				></span
+			>
+		</a>
+		<a
+			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:scale-125"
```
Comment (mtwente) [inline src/lib/components/ShareButtons.svelte:73]:
```suggestion
			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:fill-secondary-500"
```

File: src/lib/components/ShareButtons.svelte [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+						>Twitter</title
+					>
+					<path
+						d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
+					/>
+				</svg></span
+			>
+		</a>
+		<a
+			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:scale-125"
```
Comment (mtwente) [inline src/lib/components/ShareButtons.svelte:94]:
```suggestion
			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:fill-secondary-500"
```

File: src/lib/components/ShareButtons.svelte [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+						>Whatsapp</title
+					>
+					<path
+						d="M380.9 97.1C339 55.1 283.2 32 223.9 32c-122.4 0-222 99.6-222 222 0 39.1 10.2 77.3 29.6 111L0 480l117.7-30.9c32.4 17.7 68.9 27 106.1 27h.1c122.3 0 224.1-99.6 224.1-222 0-59.3-25.2-115-67.1-157zm-157 341.6c-33.2 0-65.7-8.9-94-25.7l-6.7-4-69.8 18.3L72 359.2l-4.4-7c-18.5-29.4-28.2-63.3-28.2-98.2 0-101.7 82.8-184.5 184.6-184.5 49.3 0 95.6 19.2 130.4 54.1 34.8 34.9 56.2 81.2 56.1 130.5 0 101.8-84.9 184.6-186.6 184.6zm101.2-138.2c-5.5-2.8-32.8-16.2-37.9-18-5.1-1.9-8.8-2.8-12.5 2.8-3.7 5.6-14.3 18-17.6 21.8-3.2 3.7-6.5 4.2-12 1.4-32.6-16.3-54-29.1-75.5-66-5.7-9.8 5.7-9.1 16.3-30.3 1.8-3.7 .9-6.9-.5-9.7-1.4-2.8-12.5-30.1-17.1-41.2-4.5-10.8-9.1-9.3-12.5-9.5-3.2-.2-6.9-.2-10.6-.2-3.7 0-9.7 1.4-14.8 6.9-5.1 5.6-19.4 19-19.4 46.3 0 27.3 19.9 53.7 22.6 57.4 2.8 3.7 39.1 59.7 94.8 83.8 35.2 15.2 49 16.5 66.6 13.9 10.7-1.6 32.8-13.4 37.4-26.4 4.6-13 4.6-24.1 3.2-26.4-1.3-2.5-5-3.9-10.5-6.6z"
+					/>
+				</svg></span
+			>
+		</a>
+		<a
+			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:scale-125"
```
Comment (mtwente) [inline src/lib/components/ShareButtons.svelte:116]:
```suggestion
			class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:fill-secondary-500"
```

File: src/lib/components/ShareButtons.svelte [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+			href="mailto:?subject={title}&body={title}:%20{url}"
+			aria-label="Teilen mit Email"
+		>
+			<span>
+				<svg
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 512 512"
+					aria-hidden="true"
+					class="h-6 w-6"
```
Comment (mtwente) [inline src/lib/components/ShareButtons.svelte:42]:
```suggestion
					class="h-4 w-4"
```

File: src/lib/components/ShareButtons.svelte [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+			target="_blank"
+			rel="noopener"
+			href="https://facebook.com/sharer/sharer.php?u={url}"
+			aria-label="Teilen auf Facebook"
+			><span>
+				<svg
+					aria-hidden="true"
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 320 512"
+					class="h-6 w-6"
```
Comment (mtwente) [inline src/lib/components/ShareButtons.svelte:63]:
```suggestion
					class="h-4 w-4"
```

File: src/lib/components/ShareButtons.svelte [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+			target="_blank"
+			rel="noopener"
+			href="https://twitter.com/intent/tweet?url={url}&amp;text={title}"
+			aria-label="Teilen auf Twitter"
+			><span>
+				<svg
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 512 512"
+					aria-hidden="true"
+					class="h-6 w-6"
```
Comment (mtwente) [inline src/lib/components/ShareButtons.svelte:83]:
```suggestion
					class="h-4 w-4"
```

File: src/lib/components/ShareButtons.svelte [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+			rel="noopener"
+			href="https://api.whatsapp.com/send?text={url}"
+			aria-label="Teilen auf Whatsapp"
+		>
+			<span>
+				<svg
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 448 512"
+					aria-hidden="true"
+					class="h-6 w-6"
```
Comment (mtwente) [inline src/lib/components/ShareButtons.svelte:105]:
```suggestion
					class="h-4 w-4"
```

File: src/lib/components/ShareButtons.svelte [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+			target="_blank"
+			rel="noopener"
+			href="https://www.linkedin.com/sharing/share-offsite/?url={url}"
+			aria-label="Teilen auf Linkedin"
+			><span>
+				<svg
+					aria-hidden="true"
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 448 512"
+					class="h-6 w-6"
```
Comment (mtwente) [inline src/lib/components/ShareButtons.svelte:126]:
```suggestion
					class="h-4 w-4"
```

File: src/lib/components/ShareButtons.svelte [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+				></span
+			>
+		</a>
+	</div>
+{:else}
+	<button
+		class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:scale-125"
+		on:click={handleWebShare}
+	>
+		<svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" viewBox="0 0 512 512"
```
Comment (mtwente) [inline src/lib/components/ShareButtons.svelte:141]:
```suggestion
		<svg aria-hidden="true"
		xmlns="http://www.w3.org/2000/svg"
		class="h-4 w-4"
		viewBox="0 0 512 512"
```

File: src/routes/blog/[slug]/+page.svelte [hunk 10]
```diff
@@ -28,22 +27,18 @@
...
 		<hgroup>
 			<h1 class="not-prose h2 font-bold">{data.meta.title}</h1>
-			<span class="text-sm"
-				>Veröffentlicht am {date}{#if modified !== date}, zuletzt geändert am {modified}{/if}</span
-			>
-			{#each data.meta.categories as category}
-				<a
-					class="variant-ringed badge mr-1 text-primary-500 hover:bg-primary-500 hover:text-white"
-					href="{base}/blog/kategorie/{category}">&num;{category}</a
+			<div class="flex flex-col flex-wrap gap-1 space-y-2 md:flex-row md:place-items-center">
```
Comment (mtwente) [inline src/routes/blog/[slug]/+page.svelte:30]:
```suggestion
			<div class="flex flex-col flex-wrap items-start gap-1 space-y-1 md:flex-row md:place-items-center">
```

File: src/lib/components/ShareButtons.svelte [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+					><path
+						d="M100.3 448H7.4V148.9h92.9zM53.8 108.1C24.1 108.1 0 83.5 0 53.8a53.8 53.8 0 0 1 107.6 0c0 29.7-24.1 54.3-53.8 54.3zM447.9 448h-92.7V302.4c0-34.7-.7-79.2-48.3-79.2-48.3 0-55.7 37.7-55.7 76.7V448h-92.8V148.9h89.1v40.8h1.3c12.4-23.5 42.7-48.3 87.9-48.3 94 0 111.3 61.9 111.3 142.3V448z"
+					/></svg
+				></span
+			>
+		</a>
+	</div>
+{:else}
+	<button
+		class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:scale-125"
```
Comment (mtwente) [inline src/lib/components/ShareButtons.svelte:138]:
```suggestion
		class="variant-filled-primary btn-icon btn-icon-sm fill-white hover:fill-secondary-500"
```

**Review by mtwente — CHANGES_REQUESTED** [review]
Apart from styling changes, I made a suggestion to prevent this behaviour in  `src/routes/blog/[slug]/+page.svelte`.

Did we say no to a Telegram button?

localhost_5173_blog_hag-laedt-zu-vortraegen-am-4-maerz-2024-ein (1)

**Comment by maehr** [comment]
> Apart from styling changes, I made a suggestion to prevent this behaviour in `src/routes/blog/[slug]/+page.svelte`.
> 
> Did we say no to a Telegram button?
> 
> localhost_5173_blog_hag-laedt-zu-vortraegen-am-4-maerz-2024-ein (1)

Thanks for the review and the suggestions.
PS: Yes, we won't be implementing Telegram.

**Review by mtwente — APPROVED** [review]

**Pull request merged**